src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -491,3 +491,139 @@ bool Terminal::IsCursorBlinkingAllowed() const noexcept
+// Arguments:
+// - position: viewport coordinate for selection
+// Return Value:
+// - update _selectionAnchor to new expanded location
+void Terminal::_ExpandDoubleClickSelectionLeft(const COORD position)
+{
+    // don't change the value if at/outside the boundary
+    if (position.X <= 0 || position.X >= _buffer->GetSize().RightInclusive())
+    {
+        return;
+    }
+
+    COORD positionWithOffsets = _ConvertToBufferCell(position);
+    const auto bufferViewport = _buffer->GetSize();
+    auto cellChar = _buffer->GetCellDataAt(positionWithOffsets)->Chars();
+    while (positionWithOffsets.X != 0 && !_DoubleClickDelimiterCheck(cellChar))
+    {
+        bufferViewport.DecrementInBounds(positionWithOffsets);
+        cellChar = _buffer->GetCellDataAt(positionWithOffsets)->Chars();
+    }
+
+    if (positionWithOffsets.X != 0 || _DoubleClickDelimiterCheck(cellChar))
+    {
+        // move off of delimiter to highlight properly
+        bufferViewport.IncrementInBounds(positionWithOffsets);
+    }
+
+    THROW_IF_FAILED(ShortSub(positionWithOffsets.Y, gsl::narrow<SHORT>(_ViewStartIndex()), &positionWithOffsets.Y));
+    _selectionAnchor = positionWithOffsets;
+    _selectionAnchor_YOffset = gsl::narrow<SHORT>(_ViewStartIndex());
+    _selectionActive = true;
+}
+
+// Method Description:
+// - expand the double click selection to the right (stopped by delimiter)
+// Arguments:
+// - position: viewport coordinate for selection
+// Return Value:
+// - update _endSelectionPosition to new expanded location
+void Terminal::_ExpandDoubleClickSelectionRight(const COORD position)
+{
+    // don't change the value if at/outside the boundary
+    if (position.X <= 0 || position.X >= _buffer->GetSize().RightInclusive())
+    {
+        return;
+    }
+
+    COORD positionWithOffsets = _ConvertToBufferCell(position);
+    const auto bufferViewport = _buffer->GetSize();
+    auto cellChar = _buffer->GetCellDataAt(positionWithOffsets)->Chars();
+    while (positionWithOffsets.X != _buffer->GetSize().RightInclusive() && !_DoubleClickDelimiterCheck(cellChar))
+    {
+        bufferViewport.IncrementInBounds(positionWithOffsets);
+        cellChar = _buffer->GetCellDataAt(positionWithOffsets)->Chars();
+    }
+
+    THROW_IF_FAILED(ShortSub(positionWithOffsets.Y, gsl::narrow<SHORT>(_ViewStartIndex()), &positionWithOffsets.Y));
+    _endSelectionPosition = positionWithOffsets;
+    _endSelectionPosition_YOffset = gsl::narrow<SHORT>(_ViewStartIndex());
+}
+
+// Method Description:
+// - check if buffer cell data contains delimiter for double click selection
+// Arguments:
+// - cellChar: the char saved to the buffer cell under observation
+// Return Value:
+// - true if cell data contains the delimiter.
+const bool Terminal::_DoubleClickDelimiterCheck(std::wstring_view cellChar) const
+{
+    // TODO GitHub #988: hook up delimiters to Settings
+    const std::wstring_view delimiters[] = {

```

Comment:
I know that we're going to almost immediately replace this with the settings version, but I'd be more comfortable if this was just one `wstring_view`. 🤷‍♀

Reply:
Replaced